soroban-sdk-auth/src/lib.rs

```diff
@@ -8,13 +8,48 @@ use crate::public_types::{
     SignaturePayloadV0,
 };
 
+/// Users of this module will need to pass a struct that implements NonceAuth
+/// NonceAuth should manage a nonce in the contracts storage
+///
+///
+/// Example implementation for NonceAuth -
+/// ```
+/// struct WrappedAuth(Signature);
+///
+/// impl NonceAuth for WrappedAuth {
+///
+/// fn read_nonce(e: &Env, id: Identifier) -> BigInt {
+///    let key = DataKey::Nonce(id);
+///    if let Some(nonce) = e.contract_data().get(key) {
+///       nonce.unwrap()
+///    } else {
+///        BigInt::zero(e)
+///    }
+/// }
+///
+/// fn read_and_increment_nonce(&self, e: &Env, id: Identifier) -> BigInt {
+///     let key = DataKey::Nonce(id.clone());
+///     let nonce = Self::read_nonce(e, id);
+///     e.contract_data()
+///         .set(key, nonce.clone() + BigInt::from_u32(e, 1));
+///     nonce
+/// }
+///
+/// fn get_keyed_auth(&self) -> &Signature {
+///     &self.0
+/// }
```

Comment:
Can you indent the fns so they appear formatted in the docs?

Reply:
Are you saying I should indent the whole code block? And is there a cargo command I can run to see the resulting docs?

Reply:
It just looks like the functions aren't indented one level, they're inside the impl but at the same level. Also, there's one line which has been indented by 3 characters.

Reply:
If you run `make doc` it will display the docs.

Reply:
I merged your suggestion, and updated the Makefile to build the auth docs. Thanks